### PR TITLE
Optional details matcher

### PR DESCRIPTION
**Pull request by dan-palmer**
- [x] Build a validator for `optional_details_field` validation
- [x] Apply to healthcare
- [x] Apply to risk
- [x] Remove duplicated code from the validator(pending #104)
- [x] Negative presence checks on details field(PR Feedback @whoojemaflip)

### CODE REVIEW

File: spec/support/matchers/validate_optional_details_field.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    validator = PresenceMatcher.new(details_field_method_name)
+    result = validator.matches?(subject)
+
+    set_error validator.failure_message unless result
+
+    result
+  end
+
+  def details_field_nilifies_empty_strings
+    subject.public_send("#{details_field_method_name}=", '')
```
Comment (dan-palmer) [inline spec/support/matchers/validate_optional_details_field.rb]: 👊 ❗ This bit needs to be replaced with the validator from #104.

File: spec/support/matchers/validate_optional_details_field.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    result = validator.matches?(subject)
+
+    set_error validator.error unless result
+
+    result
+  end
+
+  PresenceMatcher = Shoulda::Matchers::ActiveModel::ValidatePresenceOfMatcher
+
+  def validate_presence_of_details_field
```
Comment (whoojemaflip) [inline spec/support/matchers/validate_optional_details_field.rb]: what about negative tests?

File: spec/support/matchers/validate_optional_details_field.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  end
+
+  private
+
+  def subject
+    @subject
+  end
+
+  def reset_subject
+    @subject = @original_subject.class.new(@original_subject.model)
```
Comment (dan-palmer) [inline spec/support/matchers/validate_optional_details_field.rb:35]: 🆘 😖 🐲 Reform maintains errors across validations.

**Comment by whoojemaflip** [comment]
🍇

**Comment by cesidio** [comment]
💯

**Pull request merged**